hub-mirror/hub.py

```diff
@@ -33,36 +35,57 @@ def __init__(
         elif self.src_type == "github":
             self.src_base = 'https://api.github.com'
             self.src_repo_base = prefix + 'github.com' + suffix
+        elif self.src_type == "gitlab":
+            self.src_base = 'https://gitlab.com/api/v4'
+            self.src_repo_base = prefix + 'gitlab.com' + suffix
         self.src_repo_base = self.src_repo_base + self.src_account
         # TODO: toekn push support
         prefix = "git@" + self.dst_type + ".com:"
         self.dst_repo_base = prefix + self.dst_account
 
     def has_dst_repo(self, repo_name):
-        url = '/'.join(
-            [
-                self.dst_base, self.dst_account_type+'s', self.dst_account,
-                'repos'
-            ]
-        )
+        if self.dst_type == "gitlab":
+            url = '/'.join(
+                [
+                    self.dst_base, self.dst_account_type+'s', self.dst_account,
+                    'projects'
+                ]
+            )
+        # Gitee, Github
+        else:
+            url = '/'.join(
+                [
+                    self.dst_base, self.dst_account_type+'s', self.dst_account,
+                    'repos'
+                ]
+            )
```

Comment:
```suggestion
        # gitlab ---> projects, github/gitee ---> repos
        repo_field = "projects" if self.dst_type == "gitlab" else "repos"
        if self.dst_type == "gitlab":
            url = '/'.join(
                [
                    self.dst_base, self.dst_account_type+'s', self.dst_account,
                    repo_field
                ]
            )
```

I like this style better.